### PR TITLE
remove dependency on controller-runtime fork

### PR DESCRIPTION
**Pull request by ryandawsonuk**
depending on controller-runtime fork is awkward, as discussed at https://github.com/kubeflow/kfserving/pull/1144#discussion_r514079339

this is an attempt to remove that dependency - let's see what e2e tests say

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -26,47 +27,44 @@ require (
...
-	k8s.io/kubelet => k8s.io/kubelet v0.18.8
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.18.8
-	k8s.io/metrics => k8s.io/metrics v0.18.8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.18.8
-	k8s.io/utils => k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
-	k8s.io/test-infra => k8s.io/test-infra v0.0.0-20200803112140-d8aa4e063646
-	k8s.io/klog => k8s.io/klog v1.0.0
-	// based on https://github.com/openshift/cluster-ingress-operator/pull/429/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R34
-	sigs.k8s.io/controller-runtime => github.com/zchee/sigs.k8s-controller-runtime v0.6.1-0.20200623114430-46812d3a0a50
+	k8s.io/api => k8s.io/api v0.19.0
```
Comment (yuzisun) [inline go.mod:49]: @ryandawsonuk Did you try 0.18.6 according to last comment https://github.com/kubernetes-sigs/controller-runtime/issues/1033 conroller-runtime v0.6.x should work with that?

Comment (ryandawsonuk) [reply go.mod:49]: Weirdly trying that leads me to https://github.com/kubernetes/client-go/issues/808. I tried to work around it with replace to bring down the gnostic version but then I get the lowercase version of the error - https://github.com/kubernetes/client-go/issues/741. It seems that [1.19 does indeed fix that](https://github.com/kubernetes/client-go/issues/741#issuecomment-606883355) as this compiles with 1.19. Not looked into e2e test failures though.

**Comment by pugangxa** [comment]
I encountered the same error with this change as the e2e test during make deploy-dev:
```
The CustomResourceDefinition "inferenceservices.serving.kubeflow.org" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

**Comment by ryandawsonuk** [comment]
> I encountered the same error with this change as the e2e test during make deploy-dev:
> 
> ```
> The CustomResourceDefinition "inferenceservices.serving.kubeflow.org" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
> ```

Interesting. There seem to be some [threads](https://github.com/kubernetes-sigs/kubebuilder/issues/1140#issuecomment-561729237) out there online about that error.

**Comment by pugangxa** [comment]
> Interesting. There seem to be some [threads](https://github.com/kubernetes-sigs/kubebuilder/issues/1140#issuecomment-561729237) out there online about that error.

Yes, but maxDescLen is already set to 0 this time and it's still too long.

**Comment by pugangxa** [comment]
Think we can close this since PR #1341 fixed this issue.

**Comment by yuzisun** [comment]
thanks @pugangxa ! closing this.

**Pull request closed without merging**